### ISSUE TITLE
Error: Invalid Top N Value: N is too big for Teradata OFFSET emulation

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14346]
jOOQ generates queries like these:

```sql
select
  "v0" "ID",
  "v1" "AUTHOR_ID",
  "v2" "co_author_id",
  "v3" "DETAILS_ID",
  "v4" "TITLE",
  "v5" "PUBLISHED_IN",
  "v6" "LANGUAGE_ID",
  "v7" "CONTENT_TEXT",
  "v8" "CONTENT_PDF",
  "v9" "STATUS",
  "v10" "REC_VERSION",
  "v11" "REC_TIMESTAMP"
from (
  select *
  from (
    select top 9223372036854775807
      "test"."t_book"."ID" "v0",
      "test"."t_book"."AUTHOR_ID" "v1",
      "test"."t_book"."co_author_id" "v2",
      "test"."t_book"."DETAILS_ID" "v3",
      "test"."t_book"."TITLE" "v4",
      "test"."t_book"."PUBLISHED_IN" "v5",
      "test"."t_book"."LANGUAGE_ID" "v6",
      "test"."t_book"."CONTENT_TEXT" "v7",
      "test"."t_book"."CONTENT_PDF" "v8",
      "test"."t_book"."STATUS" "v9",
      "test"."t_book"."REC_VERSION" "v10",
      "test"."t_book"."REC_TIMESTAMP" "v11",
      row_number() over (order by "test"."t_book"."ID", "test"."t_book"."AUTHOR_ID") "rn"
    from "test"."t_book"
    order by "v0", "v1"
  ) x
) "x"
where "rn" > 1
and "rn" <= (1 + 2)
order by "rn"
```

The dummy `TOP` clause is required to avoid the error:

> Syntax error: ORDER BY is not allowed in subqueries.

But the highest possible number in `TOP` isn't `Long.MAX_VALUE`, but `999999999999999999` (it seems). Alternatively, we could generate `TOP 100 PERCENT`

**Comment by lukaseder** [comment]
Hmm, `TOP 100 PERCENT` was removed because of #13509, so this is a regression introduced by #13509

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14347), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14348)
